### PR TITLE
Add bucket histogram option

### PR DESCRIPTION
**Pull request by rzkmak**
I'm trying to add an option for bucket histogram as in the #63 issue

```
ali --buckets "1ms, 10ms, 100ms, 500ms" http://host.xz
```

But I'm not sure where I can put this part
```
0~1ms: 1 25%
1ms~10ms: 1 25%
10ms~100ms: 1 25%
100ms~500ms: 1 25%
```

Any feedback or suggestion would help me so much.
Thank you

### CODE REVIEW

**Comment by nakabonne** [comment]
@rzkmak Hey, thank you for your contribution! I can’t afford to check it, so I'll get back to you later :-)

File: main.go [hunk 1]
```diff
@@ -81,6 +83,7 @@ func parseFlags(stdout, stderr io.Writer) (*cli, error) {
 	flagSet.IntVarP(&c.connections, "connections", "c", attacker.DefaultConnections, "Amount of maximum open idle connections per target host")
 	flagSet.BoolVar(&c.noHTTP2, "no-http2", false, "Don't issue HTTP/2 requests to servers which support it.")
 	flagSet.StringVar(&c.localAddress, "local-addr", "0.0.0.0", "Local IP address.")
+	flagSet.StringVar(&c.buckets, "buckets", "", "Histogram buckets.")
```
Comment (nakabonne) [inline main.go:86]:
```suggestion
	flagSet.StringVar(&c.buckets, "buckets", "", "Histogram buckets; comma-separated list.")
```

File: main.go [hunk 2]
```diff
@@ -202,6 +206,27 @@ func validateMethod(method string) bool {
...
+
+	result := make([]time.Duration, 0)
+
+	stringBuckets := strings.Split(rawBuckets, ",")
+
+	for _, bucket := range stringBuckets {
+		trimmedBucket := strings.TrimSpace(bucket)
+		d, err := time.ParseDuration(trimmedBucket)
+		if err != nil {
+			log.Fatalf("Invalid bucket time format")
```
Comment (nakabonne) [inline main.go:222]: Can you make it give back an error if even one of them fails?

File: main.go [hunk 3]
```diff
@@ -202,6 +206,27 @@ func validateMethod(method string) bool {
...
 }
 
+func parseBucketOptions(rawBuckets string) []time.Duration {
+	if rawBuckets == "" {
+		return []time.Duration{}
+	}
+
+	result := make([]time.Duration, 0)
+
+	stringBuckets := strings.Split(rawBuckets, ",")
```
Comment (nakabonne) [inline main.go:216]:
```suggestion
	stringBuckets := strings.Split(rawBuckets, ",")
	result := make([]time.Duration, len(stringBuckets))

```

**Review by nakabonne — CHANGES_REQUESTED** [review]

**Comment by rzkmak** [comment]
thank you for the review :+1:

**Review by nakabonne — APPROVED** [review]
thank you! i will add an ability to draw the histogram buckets, so your contribution will be available after that ;)

**Pull request merged**